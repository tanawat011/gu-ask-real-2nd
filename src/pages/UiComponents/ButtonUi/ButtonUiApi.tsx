import tw from 'twin.macro'

import { Title } from 'components/PageUiComponent'
import { TableCommon } from 'components/Table'

import { ButtonApiColumns, ButtonApiData } from './data'

const TwContainer = tw.div`mb-8`

export const ButtonUiApi = () => {
  return (
    <TwContainer>
      <Title title='API' />

      <TableCommon columns={ButtonApiColumns} data={ButtonApiData} />
    </TwContainer>
  )
}
