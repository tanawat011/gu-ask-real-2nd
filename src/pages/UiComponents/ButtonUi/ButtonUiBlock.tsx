import type { LegacyRef } from 'react'
import { useEffect } from 'react'

import { useRecoilValue } from 'recoil'
import tw from 'twin.macro'

import { Button } from 'components/Button'
import { ComponentDisplay, Title } from 'components/PageUiComponent'
import { PreCode } from 'components/PreCode'
import { useDimensionChange } from 'hooks/useDimensionChange'
import { useElementDimension } from 'hooks/useElementDimension'
import { blockSelector, buttonUiAtom } from 'recoils/atoms'

const TwContainer = tw.div`mb-8`

type ButtonUiBlockProps = {
  handlePageDimension: () => Promise<void>
}

export const ButtonUiBlock: React.FC<ButtonUiBlockProps> = ({
  handlePageDimension,
}) => {
  const [ref, dimension, handleDimension] = useElementDimension()

  useDimensionChange(blockSelector, dimension)

  const { dimension: btnDms } = useRecoilValue(buttonUiAtom)

  useEffect(() => {
    handleDimension()
    handlePageDimension()
  }, [
    btnDms.variant,
    btnDms.color,
    btnDms.size,
    btnDms.shape,
    btnDms.disabled,
    btnDms.icon,
    btnDms.withIcon,
    btnDms.loading,
  ])

  return (
    <TwContainer id='block' ref={ref as LegacyRef<HTMLDivElement>}>
      <Title
        title='Block'
        description={
          <>
            Button can stretch to fill the width by add <PreCode text='block' />{' '}
            prop.
          </>
        }
      />
      <ComponentDisplay
        callback={handleDimension}
        components={
          <>
            <Button label='Non Block Button' />
            <Button label='Blocked Button' block />
            <Button label='Blocked Button' block disabled />
            <Button label='Blocked Button' block loading='Loading...' />
          </>
        }
        code={`import React from "react"

export const Component = () => {
  return (
    <>
    <Button label='Non Block Button' />
    <Button label='Blocked Button' block />
    <Button label='Blocked Button' block disabled />
    <Button label='Blocked Button' block loading='Loading...' />
    </>
  )
}`}
      />
    </TwContainer>
  )
}
