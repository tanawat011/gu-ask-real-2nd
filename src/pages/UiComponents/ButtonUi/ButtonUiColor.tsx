import type { LegacyRef } from 'react'
import { useEffect } from 'react'

import { useRecoilValue } from 'recoil'
import tw from 'twin.macro'

import { Button } from 'components/Button'
import { ComponentDisplay, Title } from 'components/PageUiComponent'
import { useDimensionChange } from 'hooks/useDimensionChange'
import { useElementDimension } from 'hooks/useElementDimension'
import { buttonUiAtom, colorSelector } from 'recoils/atoms'

const TwContainer = tw.div`mb-8`

export const ButtonUiColor = () => {
  const [ref, dimension, handleDimension] = useElementDimension()

  useDimensionChange(colorSelector, dimension)

  const { dimension: rDimension } = useRecoilValue(buttonUiAtom)

  useEffect(() => {
    handleDimension()
  }, [rDimension.variant])

  return (
    <TwContainer id='color' ref={ref as LegacyRef<HTMLDivElement>}>
      <Title
        title='Color'
        description={
          <>
            <code>twoTone</code> and <code>solid</code> button able to apply
            custom color, available colors option based on tailwind{' '}
            <code>theme.colors</code>.
          </>
        }
      />
      <ComponentDisplay
        callback={handleDimension}
        components={
          <div className='flex gap-2 items-center flex-wrap'>
            <Button label='Danger' color='rose-500' />
            <Button label='Warning' color='amber-500' />
            <Button label='Info' color='sky-500' />
            <Button label='Success' color='emerald-500' />
          </div>
        }
        code={`import React from "react"

export const Component = () => {
  return (
    <Button label='Danger' color='rose-500' />
    <Button label='Warning' color='amber-500' />
    <Button label='Info' color='sky-500' />
    <Button label='Success' color='emerald-500' />
  )
}`}
      />
    </TwContainer>
  )
}
