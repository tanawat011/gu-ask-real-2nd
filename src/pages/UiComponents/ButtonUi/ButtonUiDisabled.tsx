import tw from 'twin.macro'

import { Button } from 'components/Button'
import { ComponentDisplay, Title } from 'components/PageUiComponent'
import { PreCode } from 'components/PreCode'

import { menuList } from './data'

const TwContainer = tw.div`mb-8`

export const ButtonUiDisabled = () => {
  return (
    <TwContainer id={menuList.disabled.id}>
      <Title
        title={menuList.disabled.name}
        description={
          <>
            Disable style of the button, can set the <PreCode text='disabled' />{' '}
            prop value to <PreCode text='true' /> or <PreCode text='false' />,
            The default shape is <PreCode text='false' />.
          </>
        }
      />
      <ComponentDisplay
        components={
          <div className='flex gap-2 items-center flex-wrap'>
            <Button label='Disabled' disabled />
          </div>
        }
        code={`import React from "react"

export const Component = () => {
  return (
    <Button label='Disabled' disabled />
  )
}`}
      />
    </TwContainer>
  )
}
