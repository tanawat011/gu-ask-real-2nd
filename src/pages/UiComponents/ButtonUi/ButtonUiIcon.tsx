import tw from 'twin.macro'

import { Button } from 'components/Button'
import { IconHome } from 'components/Icons'
import { ComponentDisplay, Title } from 'components/PageUiComponent'
import { PreCode } from 'components/PreCode'

const TwContainer = tw.div`mb-8`

export const ButtonUiIcon = () => {
  return (
    <TwContainer id='icon'>
      <Title
        title='Icon'
        description={
          <>
            Button can add an icon by insert icon to <PreCode text='icon' />{' '}
            prop.
          </>
        }
      />
      <ComponentDisplay
        components={
          <div className='flex gap-2 items-center flex-wrap'>
            <Button size='xs' icon={<IconHome />} />
            <Button size='sm' icon={<IconHome />} />
            <Button size='md' icon={<IconHome />} />
            <Button size='lg' icon={<IconHome />} />
          </div>
        }
        code={`import React from "react"

export const Component = () => {
  return (
    <Button size='xs' icon={<IconHome />} />
    <Button size='sm' icon={<IconHome />} />
    <Button size='md' icon={<IconHome />} />
    <Button size='lg' icon={<IconHome />} />
    <Button size='xl' icon={<IconHome />} />
  )
}`}
      />
    </TwContainer>
  )
}
