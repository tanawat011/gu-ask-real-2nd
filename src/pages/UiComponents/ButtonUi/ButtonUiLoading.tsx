import type { LegacyRef } from 'react'
import { useEffect } from 'react'

import { useRecoilValue } from 'recoil'
import tw from 'twin.macro'

import { Button } from 'components/Button'
import { IconHome } from 'components/Icons'
import { ComponentDisplay, Title } from 'components/PageUiComponent'
import { PreCode } from 'components/PreCode'
import { useDimensionChange } from 'hooks/useDimensionChange'
import { useElementDimension } from 'hooks/useElementDimension'
import { buttonUiAtom, loadingSelector } from 'recoils/atoms'

const TwContainer = tw.div`mb-8`

type ButtonUiLoadingProps = {
  handlePageDimension: () => Promise<void>
}

export const ButtonUiLoading: React.FC<ButtonUiLoadingProps> = ({
  handlePageDimension,
}) => {
  const [ref, dimension, handleDimension] = useElementDimension()

  useDimensionChange(loadingSelector, dimension)

  const { pageDimension } = useRecoilValue(buttonUiAtom)

  useEffect(() => {
    handleDimension()
    handlePageDimension()
  }, [pageDimension])

  return (
    <TwContainer id='loading' ref={ref as LegacyRef<HTMLDivElement>}>
      <Title
        title='Loading'
        description={
          <>
            Button can set <PreCode text='loading' /> prop to{' '}
            <PreCode text='true' /> to show loading indicator, it will disabled
            button.
          </>
        }
      />
      <ComponentDisplay
        callback={handleDimension}
        components={
          <div className='flex gap-2 items-center flex-wrap'>
            <Button label='Home' loading size='xs' />
            <Button label='Home' loading />
            <Button icon={<IconHome />} label='Home' loading />
            <Button icon={<IconHome />} loading size='xs' />
            <Button icon={<IconHome />} loading />
          </div>
        }
        code={`import React from "react"

export const Component = () => {
  return (
    <Button label='Home' loading size='xs' />
    <Button label='Home' loading />
    <Button icon={<IconHome />} label='Home' loading />
    <Button icon={<IconHome />} loading size='xs' />
    <Button icon={<IconHome />} loading />
  )
}`}
      />
    </TwContainer>
  )
}
