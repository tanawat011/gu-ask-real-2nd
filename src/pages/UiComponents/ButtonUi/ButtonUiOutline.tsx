import type { LegacyRef } from 'react'
import { useEffect } from 'react'

import { useRecoilValue } from 'recoil'
import tw from 'twin.macro'

import { Button } from 'components/Button'
import { ComponentDisplay, Title } from 'components/PageUiComponent'
import { PreCode } from 'components/PreCode'
import { useDimensionChange } from 'hooks/useDimensionChange'
import { useElementDimension } from 'hooks/useElementDimension'
import { buttonUiAtom, outlineSelector } from 'recoils/atoms'

const TwContainer = tw.div`mb-8`

type ButtonUiOutlineProps = {
  handlePageDimension: () => Promise<void>
}

export const ButtonUiOutline: React.FC<ButtonUiOutlineProps> = ({
  handlePageDimension,
}) => {
  const [ref, dimension, handleDimension] = useElementDimension()

  useDimensionChange(outlineSelector, dimension)

  const { pageDimension } = useRecoilValue(buttonUiAtom)

  useEffect(() => {
    handleDimension()
    handlePageDimension()
  }, [pageDimension])

  return (
    <TwContainer id='outline' ref={ref as LegacyRef<HTMLDivElement>}>
      <Title
        title='Outline'
        description={
          <>
            Outline style of the button, can set the <PreCode text='outline' />{' '}
            prop value to <PreCode text='true' /> or <PreCode text='false' />,
            The default outline is <PreCode text='false' />.
          </>
        }
      />
      <ComponentDisplay
        callback={handleDimension}
        components={
          <div className='flex gap-2 items-center flex-wrap'>
            <Button label='Primary' variant='primary' outline />
            <Button label='Secondary' variant='secondary' outline />
            <Button label='Tertiary' variant='tertiary' outline />
            <Button label='Plain' variant='plain' outline />
            <Button label='Link' variant='link' outline />
          </div>
        }
        code={`import React from "react"

export const Component = () => {
  return (
    <Button label='Primary' variant='primary' outline />
    <Button label='Secondary' variant='secondary' outline />
    <Button label='Tertiary' variant='tertiary' outline />
    <Button label='Plain' variant='plain' outline />
    <Button label='Link' variant='link' outline />
  )
}`}
      />
    </TwContainer>
  )
}
