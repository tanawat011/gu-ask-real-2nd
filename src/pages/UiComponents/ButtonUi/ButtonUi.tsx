import { useState } from 'react'

import tw from 'twin.macro'

import { Title } from 'components/PageUiComponent'

import { Api } from './Api'
import { ButtonUiBlock } from './ButtonUiBlock'
import { ButtonUiColor } from './ButtonUiColor'
import { ButtonUiDisabled } from './ButtonUiDisabled'
import { ButtonUiIcon } from './ButtonUiIcon'
import { ButtonUiLoading } from './ButtonUiLoading'
import { ButtonUiShape } from './ButtonUiShape'
import { ButtonUiSize } from './ButtonUiSize'
import { ButtonUiVariant } from './ButtonUiVariant'
import { ButtonUiWithIcon } from './ButtonUiWithIcon'
import { RightMenuButtonUi } from './RightMenuButtonUi'

const TwContainer = tw.div`relative grid grid-cols-1 gap-8 overflow-x-auto px-8 py-6 desktop:grid-cols-5`
const TwWrapContent = tw.div`xl:col-span-4`
const TwWrapMenu = tw.div`col-span-1 hidden desktop:block`

export const ButtonUi = () => {
  const [isScrolled, setIsScrolled] = useState(false)

  const handleScroll = () => {
    setIsScrolled(true)
  }

  return (
    <>
      <TwContainer id='button-ui' onScroll={handleScroll}>
        <TwWrapContent>
          <div>
            <Title
              size='xl'
              title='Button'
              description='Button is used to triggers an action or event click'
            />
          </div>

          <ButtonUiVariant />
          <ButtonUiColor />
          <ButtonUiSize />
          <ButtonUiShape />
          <ButtonUiDisabled />
          <ButtonUiIcon />
          <ButtonUiWithIcon />
          <ButtonUiLoading />
          <ButtonUiBlock />

          <Api />
        </TwWrapContent>

        <TwWrapMenu>
          <RightMenuButtonUi isScrolled={isScrolled} />
        </TwWrapMenu>
      </TwContainer>
    </>
  )
}
