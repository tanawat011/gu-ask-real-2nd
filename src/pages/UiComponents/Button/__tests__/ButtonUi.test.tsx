import { renderWithProviders } from 'utils/jest'

import { ButtonUi } from '../ButtonUi'

jest.mock('components/PageUiComponent', () => ({
  Title: () => <div>Title</div>,
}))
jest.mock('../ButtonUiApi')
jest.mock('../ButtonUiBlock')
jest.mock('../ButtonUiDisabled')
jest.mock('../ButtonUiIcon')
jest.mock('../ButtonUiLoading')
jest.mock('../ButtonUiOutline')
jest.mock('../ButtonUiShape')
jest.mock('../ButtonUiSize')
jest.mock('../ButtonUiVariant')
jest.mock('../ButtonUiWithIcon')
jest.mock('../RightMenuButtonUi')

describe('<ButtonUi />', () => {
  test('renders correctly', () => {
    expect.assertions(1)

    const { asFragment } = renderWithProviders(<ButtonUi />)

    expect(asFragment()).toMatchSnapshot()
  })
})
