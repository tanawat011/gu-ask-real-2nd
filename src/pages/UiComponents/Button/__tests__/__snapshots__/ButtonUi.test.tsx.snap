// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonUi /> renders correctly 1`] = `
<DocumentFragment>
  .c0 {
  display: grid;
  grid-template-columns: repeat(1,minmax(0,1fr));
  gap: 2rem;
  padding-left: 2rem;
  padding-right: 2rem;
  padding-top: 1.5rem;
  padding-bottom: 1.5rem;
}

.c2 {
  grid-column: span 1 / span 1;
  display: none;
}

@media (min-width:1280px) {
  .c0 {
    grid-template-columns: repeat(5,minmax(0,1fr));
  }
}

@media (min-width:1280px) {
  .c1 {
    grid-column: span 4 / span 4;
  }
}

@media (min-width:1280px) {
  .c2 {
    display: block;
  }
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <div>
        Title
      </div>
    </div>
    <div
      class="c2"
    />
  </div>
</DocumentFragment>
`;
